lib/mercury/mercury.rb

```diff
@@ -60,17 +60,36 @@ def publish(source_name, msg, tag: '', headers: {}, &k)
     # The amqp gem caches exchange objects, so it's fine to
     # redeclare the exchange every time we publish.
     with_source(source_name) do |exchange|
-      payload = write(msg)
-      pub_opts = Mercury.publish_opts(tag, headers)
-      if publisher_confirms_enabled
-        expect_publisher_confirm(k)
-        exchange.publish(payload, **pub_opts)
-      else
-        exchange.publish(payload, **pub_opts, &k)
-      end
+      publish_internal(exchange, msg, tag, headers, &k)
+    end
+  end
+
+  ORIGINAL_TAG_HEADER = 'Original-Tag'.freeze
```

Comment:
I think 'private' constants should be grouped under the 'private' directive, 'public' constants should be grouped at the top of the class/module declaration. The style guide also emphasizes organizing in a [consistent manner].

[consistent manner]: https://github.com/bbatsov/ruby-style-guide#consistent-classes